annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastFieldElement.java

```
@@ -0,0 +1,51 @@
+package com.exactpro.epfast.annotation.processing;
+
+import com.exactpro.epfast.annotations.FastField;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.type.TypeMirror;
+
+public class FastFieldElement {
+    private Element setterMethod;
+
+    private String fieldName;
+
+    public FastFieldElement(Element setterMethod) {
+        this.setterMethod = setterMethod;
+        this.fieldName = getFastFieldName(setterMethod);
+    }
+
+    private String getFastFieldName(Element setterMethod) {
+        String fastFieldName = setterMethod.getAnnotation(FastField.class).name();
+        if (fastFieldName.isEmpty()) {
+            fastFieldName = recoverFastFieldNameFromSetter(getMethodName());
+        }
+        return fastFieldName;
+    }
+
+    static String recoverFastFieldNameFromSetter(String setterMethodName) {
+        return setterMethodName.substring(3, 4).toLowerCase() + setterMethodName.substring(4);
```

We add FastFieldElement to collection prior to validating it. You try to get name in FastFieldElement constructor. At this time you can't assume that method name is valid.

So instead of getting field name in constructor, get in getter. name can be cached there. recover...() method should throw an Exception (any RuntimeException) if name is invalid.

rename to getBeanPropertyName